File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
    static boolean[] isVisited;
    static ArrayList<Integer>[] graph;
    static int N;
    static StringBuilder sb = new StringBuilder();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());
        N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int start = Integer.parseInt(st.nextToken());
        isVisited = new boolean[N+1];
        graph = new ArrayList[N+1];
        for(int i=1; i<=N; i++) graph[i] = new ArrayList<>();

        for(int i=0; i<M; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            graph[a].add(b);
            graph[b].add(a);
        }

        for(int i=1; i<=N; i++) {
            Collections.sort(graph[i]);
        }

        isVisited[start] = true;
        DFS(start);
        sb.append("\n");
        BFS(start);
        System.out.println(sb);
        br.close();
    }

    static void BFS(int n) {
        Queue<Integer> q = new ArrayDeque<>();
        q.add(n);
        boolean[] isVisited = new boolean[N+1];
        isVisited[n] = true;

        while(!q.isEmpty()) {
            int cur = q.poll();
            sb.append(cur).append(" ");
            for(int child : graph[cur]) {
                if(!isVisited[child]) {
                    isVisited[child] = true;
                    q.add(child);
                }
            }
        }
    }

    static void DFS(int n) {
        sb.append(n).append(" ");
        for(int child : graph[n]) {
            if(!isVisited[child]) {
                isVisited[child] = true;
                DFS(child);
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

vector<vector<int>> graph;
vector<bool> isVisited;
void DFS(int node) {
    isVisited[node] = true;
    cout << node << " ";
    for(int next : graph[node]) {
        if(!isVisited[next]) DFS(next);
    }
}

void BFS(int node) {
    queue<int> q;
    q.push(node);
    isVisited[node] = true;
    
    while(!q.empty()) {
        int cur = q.front();
        q.pop();

        cout << cur << " ";
        for(int next : graph[cur]) {
            if(!isVisited[next]) {
                isVisited[next] = true;
                q.push(next);
            }
        }
    }
}

int main() {

    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int N, M, V;
    cin >> N >> M >> V;

    graph.assign(N+1, vector<int>());
    isVisited.assign(N+1, false);
    for(int i=0; i<M; i++) {
        int a, b;
        cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    for(int i=1; i<=N; i++) {
        sort(graph[i].begin(), graph[i].end());
    }

    DFS(V);
    cout << "\n";
    isVisited.assign(N+1, false);
    BFS(V);

    return 0;
}